src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -674,6 +674,16 @@ public String resolveIndexRouting(@Nullable String routing, String aliasOrIndex)
 
             aliasesOrIndices = convertFromWildcards(aliasesOrIndices, indicesOptions);
         }
+
+        if (aliasesOrIndices == null || aliasesOrIndices.length == 0) {
+            if (!indicesOptions.allowNoIndices()) {
+                throw new ElasticsearchIllegalArgumentException("Null or zero length argument for list of index or alias names "
```

Comment:
Sorry for the mixup, changes that on the original PR but picked the wrong version when moving over here.